### ISSUE TITLE
Calling #reset causes error in timer

### ISSUE THREAD

**Issue by saizai** [jfelchner/ruby-progressbar#160]
"ruby-progressbar-1.10.1/lib/ruby-progressbar/timer.rb:54:in `-': can't convert nil into an exact number (TypeError)"

https://github.com/jfelchner/ruby-progressbar/blob/043d5a8a7abc18dad4b8dc57b3cc3c8a5f701a07/lib/ruby-progressbar/timer.rb#L54

**Comment by jfelchner** [comment]
Neither `time.now` nor `started_at` should ever be `nil`.  Can you create a test that reproduces this behavior?

**Comment by wojtha** [comment]
Just got the same error. Unfortunately it is hard to isolate this error, it happens randomly and rarely.

```
 TypeError: can't convert nil into an exact number
/var/lib/gems/2.7.0/gems/ruby-progressbar-1.10.1/lib/ruby-progressbar/timer.rb:54:in `-'
/var/lib/gems/2.7.0/gems/ruby-progressbar-1.10.1/lib/ruby-progressbar/timer.rb:54:in `elapsed_seconds'
/var/lib/gems/2.7.0/gems/ruby-progressbar-1.10.1/lib/ruby-progressbar/throttle.rb:18:in `choke'
/var/lib/gems/2.7.0/gems/ruby-progressbar-1.10.1/lib/ruby-progressbar/output.rb:48:in `refresh'
/var/lib/gems/2.7.0/gems/ruby-progressbar-1.10.1/lib/ruby-progressbar/output.rb:44:in `with_refresh'
/var/lib/gems/2.7.0/gems/ruby-progressbar-1.10.1/lib/ruby-progressbar/base.rb:177:in `update_progress'
/var/lib/gems/2.7.0/gems/ruby-progressbar-1.10.1/lib/ruby-progressbar/base.rb:97:in `progress='
```

**Comment by jfelchner** [comment]
@wojtha I'd love to help here but unfortunately I'm going to need something that reproduces it. :(

**Comment by bladebhs** [comment]
Hi @jfelchner 
The following code always fails:

```ruby
require 'ruby-progressbar'

progressbar = ProgressBar.create(
  total: 100,
  format: '%a %e |%b%i| %P% Processed: %c from %C'
)

100.times { progressbar.increment; sleep(0.01) }
progressbar.reset
progressbar.total = 50
50.times { progressbar.increment; sleep(0.01) }
```

**Comment by jfelchner** [comment]
@bladebhs thank you!  This is extremely helpful.

**Comment by jfelchner** [comment]
Fixed in d1b09de thanks to @bladebhs!  It'll go out in the next version.
